### ISSUE TITLE
NullPointer on MacroProcessor with forEach

### ISSUE THREAD

**Issue by hypery2k** [galenframework/galen#303]
When trying to use the forEach loop I'm getting the following error:
```
INFO: Running Galen check layout
java.lang.NullPointerException
	at com.galenframework.speclang2.reader.pagespec.MacroProcessor.process(MacroProcessor.java:61)
	at com.galenframework.speclang2.reader.pagespec.MacroProcessor$1.visitLoop(MacroProcessor.java:178)
	at com.galenframework.speclang2.reader.pagespec.ForLoop.apply(ForLoop.java:227)
```
This is my spec:
```python
@import shared/common.spec

@objects
  js-code-snippet-*		div.bs-callout
  ad-container			#carbonads-container
  ad					.carbonad

# common layout checks
@import shared/commonLayout.spec

# concrete layout tests

= Code snippets should be vertical aligned =

  @forEach [js-code-snippet-*] as snippet, next as nextSnippet
  ${snippet}:
    aligned vertically left ${nextSnippet}
    
= Ad should be centered =

  ad:
    centered horizontally on ad-container

```
If I change the spec, it works: 
```python
@import shared/common.spec

@objects
  js-code-snippet-*		div.bs-callout
  ad-container			#carbonads-container
  ad					.carbonad

# common layout checks
@import shared/commonLayout.spec

# concrete layout tests

= Code snippets should be vertical aligned =

  js-code-snippet-*:
    aligned vertically left js-code-snippet-1
    
= Ad should be centered =

  ad:
    centered horizontally on ad-container

```

**Comment by ishubin** [comment]
That is because you didn't indent the entries below the forEach statement.
Instead of
```
@forEach [js-code-snippet-*] as snippet, next as nextSnippet
${snippet}:
    aligned vertically left ${nextSnippet}
```
You should define
```
  @forEach [js-code-snippet-*] as snippet, next as nextSnippet
      ${snippet}:
          aligned vertically left ${nextSnippet}
```

But I will fix this nullpointer exception as it is a bug anyways

**Comment by hypery2k** [comment]
thanks thats working!

**Comment by hypery2k** [comment]
fixed with 2.0.7
